markers/markers.go

```diff
@@ -150,62 +163,9 @@ func If(err error, pred func(err error) (interface{}, bool)) (interface{}, bool)
 // package location or a different type, ensure that
 // RegisterTypeMigration() was called prior to IsAny().
 func IsAny(err error, references ...error) bool {
-	if err == nil {
-		for _, refErr := range references {
-			if refErr == nil {
-				return true
-			}
-		}
-		// The mark-based comparison below will never match anything if
-		// the error is nil, so don't bother with computing the marks in
-		// that case. This avoids the computational expense of computing
-		// the reference marks upfront.
-		return false
-	}
-
-	// First try using direct reference comparison.
-	for c := err; c != nil; c = errbase.UnwrapOnce(c) {
-		for _, refErr := range references {
-			if refErr == nil {
-				continue
-			}
-			isComparable := reflect.TypeOf(refErr).Comparable()
-			if isComparable && c == refErr {
-				return true
-			}
-			// Compatibility with std go errors: if the error object itself
-			// implements Is(), try to use that.
-			if tryDelegateToIsMethod(c, refErr) {
-				return true
-			}
-		}
-
-		// Recursively try multi-error causes, if applicable.
-		for _, me := range errbase.UnwrapMulti(c) {
-			if IsAny(me, references...) {
-				return true
-			}
-		}
-	}
-
-	// Try harder with marks.
-	// Note: there is a more effective recursive algorithm that ensures
-	// that any pair of string only gets compared once. Should this
-	// become a performance bottleneck, that algorithm can be considered
-	// instead.
-	refMarks := make([]errorMark, 0, len(references))
-	for _, refErr := range references {
-		if refErr == nil {
-			continue
-		}
-		refMarks = append(refMarks, getMark(refErr))
-	}
-	for c := err; c != nil; c = errbase.UnwrapOnce(c) {
-		errMark := getMark(c)
-		for _, refMark := range refMarks {
-			if equalMarks(errMark, refMark) {
-				return true
-			}
+	for _, reference := range references {
+		if Is(err, reference) {
```

Comment:
I think the reason we didn't use this implementation originally is `Is` was pretty slow. So it was better to use the quick check before the slow mark check. Now that `Is` is efficient, we don't need to duplicate the implementations.